### ISSUE TITLE
Improve 1d binning plots

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/xdem#544]
Currently, the function `xdem.spatialstats.plot_1d_binning` only shows one metric at a time, with a scatter plot.

It would be useful to at least show a central tendency + dispersion (e.g. median + NMAD) as a barplot. Ideally, we could even make a violin plot like Fig 4 in Hugonn et al. (2022), but this would probably need to be a separate function as it requires the full array of values.

**Comment by rhugonnet** [comment]
I like the summary plot idea :smile:
All of these should be relatively small effort

**Comment by adebardo** [comment]
This will be helpfull for a nice report with the CLI, we must detail this ticket :)

**Comment by rhugonnet** [comment]
Nice, you found the old issue!

Our plan was to use the metadata stored about binning+fitting to generate any plot consistently, whether it's `NuthKaab`, `TerrainBias` or other.
For any method that uses binning and/or fitting, we already store all inputs and outputs required for the plots in the two typed dict here: https://github.com/GlacioHack/xdem/blob/aad0d2c5a49769d4c4c02b1f4359fb5f153fd392/xdem/coreg/base.py#L1645

This would allow to generate any plot the same way (inspiring ourselves from `plot_1d_binning` and `plot_2d_binning` in `spatialstats`?), but this time as a `Coreg` method that automatically detects the number of dimensions (stored in the metadata `nd`).
1. The sparse binning can be plotted using the `bin_dataframe` output, with names corresponding to `bias_var_names`.
2. Then, the function fit and later applied can be showed on top of the binning. 
If it is parametric, we can simply call the `fit_func` and `fit_params` to plot it as here: https://github.com/GlacioHack/xdem/blob/aad0d2c5a49769d4c4c02b1f4359fb5f153fd392/xdem/coreg/biascorr.py#L279
If it is empirical, we can simply mirror the interpolated/per-bin function defined here:
https://github.com/GlacioHack/xdem/blob/aad0d2c5a49769d4c4c02b1f4359fb5f153fd392/xdem/coreg/biascorr.py#L285

One other thing we could add for better visualization would be a small subsample of original samples (often elevation changes) that we could keep in memory (e.g. 10,000 points?) to give an idea of the distribution in the background.

Looking at the process again, the only limiting thing would be that, because `Affine` methods are iterative, we don't store that metadata the same way right now... (would need to be saved at each iteration). This might a bit of work to add... I can think of the structure for this in #759. (as I'm already adding other iterative metadata in there)

**Comment by adehecq** [comment]
> This will be helpfull for a nice report with the CLI, we must detail this ticket :)

Yes, I would love to have these plots generated automatically by the workflow "diff-analysis" 😄
